### DISCUSSION TITLE
Is it possible to connect other payment gateway not in the list?

### DISCUSSION THREAD

**Discussion by curiouscod3** [swellstores/community#6] [Help] [answered]
I want to use Swell but I have a trouble with your payment options.
I don't see a payment gateway option that I want to connect to Swell. 
Is it possible to connect other payment gateways not in the list(such as Stripe and Paypal)?
or do you have a plan to support a custom payment connection feature in the future?

For example, Shopify supports tons of payment providers available so that I can easily find alternative method even incl. cryptocurrency payment providers.
https://www.shopify.com/payment-gateways/south-korea

**Comment by logeshswell** [comment, marked as answer]
@curiouscod3 Brilliant question! Yes, it is possible to connect a custom payment gateway with Swell.
Please follow the below steps to implement the same:
1. Create a manual payment method. (eg: Airpay)
2. In your custom checkout, pass the customer-inputted card details to your backend.
3. In your backend, create an airpay payment order and send the card details for making the payment.
4. If the payment is successful,
  a. Submit the cart and place an order.
  b. Set `order.billing.method` as airpay id, `order.payment_marked: true` and `order.billing.card.token` as airpay payment token.
  c. Return 200 response with message `success` to the checkout.
5. Else
  a. Return 400 response with payment error description to the checkout.

**Reply by curiouscod3** [reply]
So `order.billing.method` and `order.payment_marked: true` and `order.billing.card.token` are all to set up for a custom payment option?
I get it but It would be nice with an example project with NextJS/React for instance:) 
Is it applied for other payment options like web3 crypto(such as coinbain) as well?

Will you update the official documentation in detail for this later?

**Reply by logeshswell** [reply]
No, those 3 fields should be updated for the placed order object.
I will definitely come up with an example project with detailed documentation for this use case and link it to this discussion sometime later. We're also working on similar ideas parallelly.

> Is it applied for other payment options like web3 crypto(such as coinbain) as well?

This solution applies to any of the custom payment methods, including web3 crypto as you mentioned.

**Reply by curiouscod3** [reply]
Cool.. I will look forward to it ! We need  proper examples

**Comment by woutercouvaras** [comment]
Hi @logeshswell,

This is a helpful thread.  I have a quick question that relates to this line:

> In your backend, create an airpay payment order and send the card details for making the payment.

Does "airpay payment" mean a swell '[Payment](https://developers.swell.is/frontend-api/payments)' as per swell.js or as per the [Backend API](https://developers.swell.is/backend-api/payments/create-a-payment#create-a-payment)?

We use Paystack (the Stripe of Africa) and we're trying to plug it in as a custom gateway.  From my intial bit of reading and playing around it looks like I might need to use the backend api, which I'm about to try, but if you have any additional info that you can share about the process/flow of creating a custom payment, I'd really appreciate it!

**Reply by logeshswell** [reply]
@woutercouvaras No, airpay payment order refers to the [Airpay order API](https://sanctum.airpay.co.in/#api-INTEGRATION-DirectIndex).
In your case, it would be [initiate transaction API](https://paystack.com/docs/api/transaction/#initialize) in Paystack.

**Reply by woutercouvaras** [reply]
Thanks, I've already managed to get a custom payment working :D

**Comment by logeshswell** [comment]
Please follow this guide for detailed explanation to setup a custom payment gateway in Swell.
https://github.com/orgs/swellstores/discussions/255#discussioncomment-5960624

**Reply by logeshswell** [reply]
cc: @curiouscod3 @woutercouvaras
